test/prefer-number-is-integer.mjs

```diff
@@ -0,0 +1,108 @@
+import {getTester} from './utils/test.mjs';
+
+const {test} = getTester(import.meta);
+
+const suggestionCase = ({code, suggestions}) => ({
+	code,
+	errors: [
+		{
+			messageId: 'preferNumberIsInteger',
+			suggestions,
+		},
+	],
+});
+
+test({
+	valid: [
+		'Number.isInteger(13)',
+		'Number.isInteger(13.0)',
+		'Number.isInteger(value)',
+		'(value^0) === notValue',
+		'(value | 0) === notValue',
+		'Math.round(value) === notValue',
+		'parseInt(value, 10) === notValue',
+		'~~value === notValue',
+	],
+	invalid: [
+		suggestionCase({code: 'value % 1 === 0', suggestions: [{
+			messageId: 'preferNumberIsIntegerSuggestion',
+			output: 'Number.isInteger(value)',
+		}]}),
+		suggestionCase({code: 'value % 1 == 0', suggestions: [{
+			messageId: 'preferNumberIsIntegerSuggestion',
+			output: 'Number.isInteger(value)',
+		}]}),
+		suggestionCase({code: '(value^0) === value', suggestions: [{
+			messageId: 'preferNumberIsIntegerSuggestion',
+			output: 'Number.isInteger(value)',
+		}]}),
+		suggestionCase({code: '(value^0) == value', suggestions: [{
+			messageId: 'preferNumberIsIntegerSuggestion',
+			output: 'Number.isInteger(value)',
+		}]}),
+		suggestionCase({code: '(value | 0) === value', suggestions: [{
+			messageId: 'preferNumberIsIntegerSuggestion',
+			output: 'Number.isInteger(value)',
+		}]}),
+		suggestionCase({code: '(value | 0) == value', suggestions: [{
+			messageId: 'preferNumberIsIntegerSuggestion',
+			output: 'Number.isInteger(value)',
+		}]}),
+		suggestionCase({code: 'parseInt(value, 10) === value', suggestions: [{
+			messageId: 'preferNumberIsIntegerSuggestion',
+			output: 'Number.isInteger(value)',
+		}]}),
+		suggestionCase({code: 'parseInt(value, 10) == value', suggestions: [{
+			messageId: 'preferNumberIsIntegerSuggestion',
+			output: 'Number.isInteger(value)',
+		}]}),
```

Comment:
I don't think this should be fixed

```js
parseInt('11', 10) == '11'
true
Number.isInteger('11')
false
````

But it's safe if it's `===`, maybe we should only check `===`

Reply:
Sorry, I didn't see that this is a suggestion.